base_notify_mattermost.yml

```diff
@@ -9,8 +8,27 @@
   stage: Notify Finished Deployment
   variables:
     MATTERMOST_CHANNEL: ${MATTERMOST_CHANNEL:-gitlab/notifications}
+
+.notify_mattermost_dev:
+  except:
+    refs:
+      - tags
+      - master
+  environment:
+    name: review/${CI_COMMIT_REF_NAME}-dev
+  variables:
+    MATTERMOST_CHANNEL: ${MATTERMOST_CHANNEL:-gitlab/notifications}
+
+.notify_mattermost_prod:
+  only:
+    refs:
+      - master
+  environment:
+    name: prod
+  variables:
+    MATTERMOST_CHANNEL: ${MATTERMOST_CHANNEL:-gitlab/notifications}
   before_script:
-    - HOST="https://${CI_ENVIRONMENT_SLUG}-${CI_PROJECT_NAME}.${KUBE_INGRESS_BASE_DOMAIN}"
+    - HOST="https://${CI_PROJECT_NAME}.${KUBE_INGRESS_BASE_DOMAIN}"
```

Comment:
@revolunet This is `autodevops_simple_app` centric as it's using the `autodevops_simple_app` env name, urls and only/except 
I think we should put this outside of the `notify_mattermost` (in autodevops like the github notify and/or the documentation) or prefix it with the env/only/expect goal